platform/bitcoin/client.go

```diff
@@ -14,36 +17,45 @@ type Client struct {
 }
 
 func (c *Client) GetTransactions(address string) (TransactionsList, error) {
-	var transfers TransactionsList
+	var txs TransactionsList
 
-	path := fmt.Sprintf("address/%s", address)
-	err := client.Request(c.HTTPClient, c.URL, path, url.Values{"details": {"txs"}}, &transfers)
+	path := fmt.Sprintf("v2/address/%s", address)
+	query := url.Values{
+		"pageSize": {strconv.FormatInt(blockatlas.TxPerPage*4, 10)},
+		"details":  {"txs"},
+	}
+	err := client.Request(c.HTTPClient, c.URL, path, query, &txs)
 
-	return transfers, err
+	return txs, err
 }
 
 func (c *Client) GetTransactionsByXpub(xpub string) (TransactionsList, error) {
-	var transfers TransactionsList
+	var txs TransactionsList
 
 	path := fmt.Sprintf("v2/xpub/%s", xpub)
-	err := client.Request(c.HTTPClient, c.URL, path, url.Values{"details": {"txs"}}, &transfers)
+	query := url.Values{
+		"pageSize": {strconv.FormatInt(blockatlas.TxPerPage*4, 10)},
+		"details":  {"txs"},
+		"tokens":   {"derived"},
+	}
+	err := client.Request(c.HTTPClient, c.URL, path, query, &txs)
 
-	return transfers, err
+	return txs, err
 }
 
-func (c *Client) GetTransactionReceipt(id string) (*TransferReceipt, error) {
+func (c *Client) GetTransactionById(id string) (*Transaction, error) {
 	url := fmt.Sprintf("%s/v2/tx/%s", c.URL, id)
 	resp, err := c.HTTPClient.Get(url)
 	if err != nil {
 		return nil, err
 	}
 	defer resp.Body.Close()
 
-	var receipt TransferReceipt
-	err = json.NewDecoder(resp.Body).Decode(&receipt)
+	var transaction Transaction
```

Comment:
take a look at waves clients, how to send and handle requests.

Reply:
We can always improve, here I just renamed variables. BTW, you should do a careful review earlier, the namings were nonsense, get txs by address didn't work too